app/main.py

```diff
@@ -1,13 +1,35 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors":
+            [
+                format_linter_error(error) for error in errors
+            ],
+        "path": file_path,
+        "status": "failed" if ["errors"] != [] else "success"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": linter_report["./test_source_code_2.py"],
+            "path": "./test_source_code_2.py",
+            "status": "passed",
+        },
+        format_single_linter_file("./source_code_2.py",
```

Comment:
use loop and don't use code like this `"./source_code_2.py"`, you should get data from `linter_report`